Packet++/src/TcpLayer.cpp

```diff
@@ -29,7 +29,9 @@ TcpOptionData* TcpLayer::getTcpOptionData(TcpOption option)
 		return NULL;
 
 	uint8_t* curOptPtr = m_Data + sizeof(tcphdr);
-	while ((curOptPtr - m_Data) < m_DataLen)
+	uint16_t dataOffset = ((tcphdr *)m_Data)->dataOffset * 4;
+
+	while ((curOptPtr - m_Data) < dataOffset)

```

Comment:
Thanks for the fix! I think it's worth checking both m_DataLen and dataOffset in case one of them is falsely set. What do you thin?